app/src/main/java/org/simple/clinic/facility/alertchange/AlertFacilityChangeSheet.kt

```diff
@@ -30,21 +33,47 @@ class AlertFacilityChangeSheet : BottomSheetActivity() {
   companion object {
     const val FACILITY_CHANGE = 101
     private const val CURRENT_FACILITY_NAME = "current_facility"
+    private const val CONTINUE_TO_SCREEN = "continue_to_screen"
+
+    const val EXTRA_CONTINUE_SCREEN = "extra_continue_screen"
 
     fun intent(context: Context, currentFacilityName: String): Intent {
       val intent = Intent(context, AlertFacilityChangeSheet::class.java)
       intent.putExtra(CURRENT_FACILITY_NAME, currentFacilityName)
       return intent
     }
+
+    fun intentForScreen(
+        context: Context,
+        currentFacilityName: String,
+        continueToScreen: FullScreenKey
+    ): Intent {
+      val intent = Intent(context, AlertFacilityChangeSheet::class.java)
+      intent.putExtra(CURRENT_FACILITY_NAME, currentFacilityName)
+      intent.putExtra(CONTINUE_TO_SCREEN, continueToScreen)
+      return intent
+    }
+
+    fun <T : Parcelable> readExtra(intent: Intent): T {
+      return intent.getParcelableExtra(EXTRA_CONTINUE_SCREEN) as T
+    }
   }
 
   private val currentFacilityName by lazy {
     intent.getStringExtra(CURRENT_FACILITY_NAME)!!
   }
 
+  private val continueToScreen by lazy {
+    intent.getParcelableExtra(CONTINUE_TO_SCREEN) as FullScreenKey
+  }
```

Comment:
These could be `unsafeLazy` since they will only be invoked from the UI thread, so we can avoid the synchronisation penalty.